src/main/java/seedu/apollo/command/task/AddCommand.java

```diff
@@ -149,25 +154,47 @@ public void execute(TaskList taskList, Ui ui, Storage storage, ModuleList module
      *
      * @param taskList The TaskList to be added to.
      * @throws DateTimeParseException If any date is entered in the wrong format.
-     * @throws DateOverException If any date of the task occurs before the current date.
-     * @throws DateOrderException If an Event's end date occurs before its start date.
-     * @throws UnexpectedException If the command stored is not recognised.
+     * @throws DateOverException      If any date of the task occurs before the current date.
+     * @throws DateOrderException     If an Event's end date occurs before its start date.
+     * @throws UnexpectedException    If the command stored is not recognised.
      */
-    private void addTask(TaskList taskList)
+    private void addTask(TaskList taskList, Ui ui)
             throws DateTimeParseException, DateOverException, DateOrderException, UnexpectedException {
         switch (command) {
         case COMMAND_TODO_WORD:
             taskList.add(new ToDo(desc));
             break;
         case COMMAND_DEADLINE_WORD:
+
             taskList.add(new Deadline(desc, by));
             break;
         case COMMAND_EVENT_WORD:
-            taskList.add(new Event(desc, from, to));
+            Event event = new Event(desc, from, to);
+            if (isClashingEvent(taskList, event.getToDate(), event.getFromDate())) {
+                ui.printClashingEventMessage();
+            }
+            taskList.add(event);
             break;
         default:
             throw new UnexpectedException("Adding Task");
         }
     }
 
+
+    public boolean isClashingEvent(TaskList taskList, LocalDateTime from, LocalDateTime to) {
+        for (Task task : taskList) {
+            if (task instanceof Event) {
+                Event event = (Event) task;
+                if ((event.getFromDate().isAfter(to) || event.getToDate().isBefore(from))) {
+                    continue;
+                }
+
+                return true;
+            }
+        }
+        return false;
```

Comment:
Would suggest further abstraction, so you use your event date to find a list of events which fall between start and end date, and then you iterate through this list. Iterating through the entire task list could be computationally expensive